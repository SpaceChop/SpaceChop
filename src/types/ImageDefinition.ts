import * as t from 'runtypes';
import Format from './Format';
import ImageFaceBox from './ImageFaceBox';
import PositiveNumber from './PositiveNumber';

const ImageDefinition = t.Record({
  width: PositiveNumber,
  height: PositiveNumber,
  type: Format,
}).And(t.Partial({
  alpha: t.Boolean,
  interlacing: t.Boolean,

  // Is only set if facedetection is done
  // ie. when an operation requries centering on face
  faces: t.Array(ImageFaceBox),

  animated: t.Boolean,
  size: t.Number,
  lossy: t.Boolean,
}));

export type ImageDefinition = t.Static<typeof ImageDefinition>;
export default ImageDefinition;
