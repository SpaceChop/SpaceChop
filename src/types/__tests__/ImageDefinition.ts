import ImageDefinition from '../ImageDefinition';

describe('validate ImageDefinition', () => {
  describe('invalid', () => {
    it('no empty config', () => {
      const config = {};
      expect(() => ImageDefinition.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('faulty width', () => {
      const config = {
        width: -1,
        heigth: 100,
        type: 'jpeg',
      };
      expect(() => ImageDefinition.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('faulty height', () => {
      const config = {
        width: 100,
        heigth: -1,
        type: 'jpeg',
      };
      expect(() => ImageDefinition.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('faulty type', () => {
      const config = {
        width: 100,
        heigth: 100,
        type: 'jpg',
      };
      expect(() => ImageDefinition.check(config)).toThrowErrorMatchingSnapshot();
    });
  });
});
