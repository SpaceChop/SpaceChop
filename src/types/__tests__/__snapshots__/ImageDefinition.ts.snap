// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate ImageDefinition invalid faulty height 1`] = `"Expected number, but was undefined"`;

exports[`validate ImageDefinition invalid faulty type 1`] = `"Expected number, but was undefined"`;

exports[`validate ImageDefinition invalid faulty width 1`] = `"please no negative number"`;

exports[`validate ImageDefinition invalid no empty config 1`] = `"Expected number, but was undefined"`;
